services/wallet/router/router_v2.go

```diff
@@ -51,6 +50,22 @@ type RouteInputParams struct {
 	Username  string       `json:"username"`
 	PublicKey string       `json:"publicKey"`
 	PackID    *hexutil.Big `json:"packID"`
+
+	// Used for tests only
+	TestsMode  bool
+	TestParams *RouterTestParams
+}
+
+type RouterTestParams struct {
```

Comment:
Is there any way we could have a test processor that we would inject, having an API like:

```
funct setProcessor() {
}
```
or
```
func setTestMode() {
}
```
, and instead of of doing
if testMode in code we would leverage the injected test element


Note: we could do same for fees and inject a test fees manager?

Reply:
Yes, this is something I was thinking of, but couldn't figure out what would be the best approach to achieve the same what I currently have for tests? I mean, I can mock all types that implement `PathProcessor` and that should work, but even then I need to have that `if testsMode` to properly set up processors, or I am not entirely sure how to do that, not sure if I can do that with making an interface for the router and then `mockgen`?